tap_github/streams.py

```diff
@@ -0,0 +1,713 @@
+                            if bookmark_dttm >= min_bookmark_value:
+
+                                if self.tap_stream_id in selected_stream_ids:
+                                    rec = transformer.transform(record, stream_catalog['schema'], metadata=metadata.to_map(stream_catalog['metadata']))
+
+                                    singer.write_record(self.tap_stream_id, rec, time_extracted=extraction_time)
+                                    counter.increment()
+
+                                for child in self.children:
+                                    if child in stream_to_sync:
+
+                                        parent_id = tuple(parent_record.get(key) for key in STREAMS[child]().id_keys)
+
+                                        self.get_child_records(client,
+                                                            catalog,
+                                                            child,
+                                                            parent_id,
+                                                            repo_path,
+                                                            state,
+                                                            start_date,
+                                                            parent_record.get(self.replication_keys),
+                                                            stream_to_sync,
+                                                            selected_stream_ids,
+                                                            parent_record = parent_record)
+                        else:
+                            LOGGER.warning("Skipping this record for %s stream with %s = %s as it is missing replication key %s.",
+                                        self.tap_stream_id, self.key_properties, record[self.key_properties], self.replication_keys)
+
+
+            # Write bookmark for incremental stream.
+            self.write_bookmarks(self.tap_stream_id, selected_stream_ids, max_bookmark_value, repo_path, state)
+
+        return state
+
+class IncrementalOrderedStream(Stream):
+
+    def sync_endpoint(self,
+                      client,
+                      state,
+                      catalog,
+                      repo_path,
+                      start_date,
+                      selected_stream_ids,
+                      stream_to_sync
+                      ):
+        """
+        A sync function for streams that have records in the descending order of replication key value. For such streams,
+        iterate only the latest records.
+        """
+        bookmark_value = get_bookmark(state, repo_path, self.tap_stream_id, "since", start_date)
+        current_time = datetime.today().strftime('%Y-%m-%dT%H:%M:%SZ')
+
+        max_bookmark_value = self.get_min_bookmark(self.tap_stream_id, selected_stream_ids, current_time, repo_path, start_date, state)
```

Comment:
@prijendev Is the variable name reflects the correct meaning? You are getting the minimum bookmark value, but storing it in `max_bookmark_value`

Reply:
Updated variable name to `min_bookmark_value`.